sunpy/instr/aia/response.py

```diff
@@ -0,0 +1,462 @@
+                dict['filtersincludemesh'] = values['filtersincludemesh']
+                dict['meshtrans'] = values['meshtrans']
+                dict['usecontam'] = values['usecontam']
+                dict['contamthick'] = values['contamthick']
+                dict['usephottoelec'] = values['usephottoelec'] * u.photon/u.electron
+                dict['elecperev'] = values['elecperev'] * (u.electron /u.ct) # u.eV?
+                dict['usephottodn'] = values['usephottodn'] * u.photon / u.ct
+                dict['elecperdn'] = values['elecperdn'] * (u.electron/u.ct)  # the ccd gain!!
+                dict['useerror'] = values['useerror']
+                dict['fp_filter'] = values['fp_filter'] * u.cm ** 2
+                dict['ent_filter'] = values['ent_filter'] * u.cm ** 2
+                dict['primary'] = values['primary'] * u.cm ** 2
+                dict['secondary'] = values['secondary'] * u.cm ** 2
+                dict['ccd'] = values['ccd'] * u.cm**2
+
+                # indexing the next two series did not return expected results because not all have len 6
+                index = self.wavelength_centers.index(int(wavelength))
+                dict['contam'] = values['contam'][0] * u.cm**2
+                dict['cross_area'] = values['cross_area'][index]
+
+
+                assert len(dict) != 0
+                return dict
+
+    def effective_area(self, wavelength,  compare=False):
+        """
+        AIA instrument response / effective area
+        - contains information about the efficiency of the telescope optics.
+        - trying to reformat effective area structure, convert to DN units
+
+        Effective Area = Geo_Area * Reflectance of Primary and Secondary Mirrors * Transmission Efficiency of filters * Quantum Efficiency of CCD * Correction for time-varying contamination
+
+        A_{eff}=A_{geo}R_p(\lambda)R_S(\lambda)T_E(\lambda)T_F(\lambda)D(\lambda,t)Q(\lambda)
+        site github ^^^     @
+
+        Parameters
+        ----------
+        wavelength: int
+            the wavelength center of the channel of interest
+
+        Variables:
+        ----------
+        Reflectance = The combined reflectance of the primary and secondary mirrors
+        Transmission_efficiency = for the focal plane and entrence filters
+
+        Returns
+        -------
+        effective_area: np.array
+            if compare is True: returns the effective area loaded in from the ssw genx file.
+            if compare is False: returns the calculations here from Boerner
+
+
+        """
+
+        assert wavelength in self.wavelength_centers, "Choose one integer value from wavelength centers: ' + str(self.wavelength_centers) or 'all'"
+
+        # define variables to call from channel dictionary
+        var = self.property_per_channel_dict(wavelength)
+        Reflectance = var['primary'] * var['secondary']
+        Transmission_efficiency = var['fp_filter'] * var['ent_filter']
```

Comment:
Agreed. We should try to use much more descriptive keys in these dataframs/dictionaries than those used by the IDL data structures.